### PR TITLE
[BugFix] fix blocked scan and add more logs to check blocked drivers

### PR DESCRIPTION
**Pull request by fzhedu**
Fixes #issue

1. scan operator is blocked if one dirver don't submit tasks and waiting until timeout, as it can't get tokens which is holded by other dirvers in the same fragment; so this PR let finishing scan operators release tokens in time.
2. add more logs to check blocked drivers.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exec/pipeline/query_context.h [hunk 1]
```diff
@@ -95,7 +95,7 @@ class QueryContext : public std::enable_shared_from_this<QueryContext> {
     void set_enable_profile() { _enable_profile = true; }
     bool enable_profile() { return _enable_profile; }
     void set_runtime_profile_report_interval(int64_t runtime_profile_report_interval_s) {
-        _runtime_profile_report_interval_ns = 1'000'000'000L * runtime_profile_report_interval_s;
```
Comment (liuyehcf) [inline be/src/exec/pipeline/query_context.h:98]: why LL?

**Review by liuyehcf — DISMISSED** [review]

File: be/src/exec/pipeline/query_context.h (hunk 1 above)
Comment (fzhedu) [reply be/src/exec/pipeline/query_context.h:98]: avoid overflow L

File: be/src/exec/pipeline/scan/scan_operator.cpp [hunk 2]
```diff
@@ -217,8 +216,20 @@ void ScanOperator::_detach_chunk_sources() {
 
 Status ScanOperator::set_finishing(RuntimeState* state) {
     std::lock_guard guard(_task_mutex);
+    if (UNLIKELY(state != nullptr && state->query_ctx()->is_query_expired() &&
+                 (_num_running_io_tasks > 0 || _submit_task_counter->value() == 0 ||
+                  _peak_io_tasks_counter->value() == 0))) {
+        LOG(WARNING) << "set_finishing scan fragment " << print_id(state->fragment_instance_id()) << " driver_id  "
```
Comment (stdpain) [inline be/src/exec/pipeline/scan/scan_operator.cpp:222]: prefer VLOG

File: be/src/runtime/data_stream_mgr.cpp [hunk 3]
```diff
@@ -151,6 +151,8 @@ Status DataStreamMgr::transmit_chunk(const PTransmitChunkParams& request, ::goog
         // in acquiring _lock.
         // TODO: Rethink the lifecycle of DataStreamRecvr to distinguish
         // errors from receiver-initiated teardowns.
+        LOG(WARNING) << request.sender_id() << " sender transmits chunks to a non-existing receiver fragment "
```
Comment (stdpain) [inline be/src/runtime/data_stream_mgr.cpp:154]: prefer VLOG

File: be/src/runtime/sender_queue.cpp [hunk 4]
```diff
@@ -661,6 +667,8 @@ Status DataStreamRecvr::PipelineSenderQueue::add_chunks(const PTransmitChunkPara
     DCHECK(!(keep_order && use_pass_through));
     DCHECK(request.chunks_size() > 0 || use_pass_through);
     if (_is_cancelled || _num_remaining_senders <= 0) {
+        LOG(WARNING) << print_id(request.finst_id()) << " adds chunks to "
```
Comment (stdpain) [inline be/src/runtime/sender_queue.cpp:670]: It's a normal case. prefer VLOG

File: be/src/runtime/sender_queue.cpp [hunk 5]
```diff
@@ -535,6 +537,11 @@ void DataStreamRecvr::PipelineSenderQueue::decrement_senders(int be_number) {
     VLOG_FILE << "decremented senders: fragment_instance_id=" << print_id(_recvr->fragment_instance_id())
               << " node_id=" << _recvr->dest_node_id() << " #senders=" << _num_remaining_senders
               << " be_number=" << be_number;
+    if (UNLIKELY(_num_remaining_senders < 0)) {
```
Comment (stdpain) [inline be/src/runtime/sender_queue.cpp:540]: VLOG is enough in 537

File: be/src/exec/pipeline/scan/scan_operator.cpp (hunk 2 above)
Comment (fzhedu) [reply be/src/exec/pipeline/scan/scan_operator.cpp:222]: if without this log, we can't know the scan is running or not run any more. The profile may not enable or lost, the only way is to get some msg from log.
It only works when timeout to help us to fix bugs.

File: be/src/exec/pipeline/exchange/sink_buffer.cpp [hunk 6]
```diff
@@ -194,25 +194,36 @@ int64_t SinkBuffer::_network_time() {
...
         _is_finishing = true;
-        if (_total_in_flight_rpc > 0) {
-            std::stringstream ss;
-            auto remain_rpc_num = 0;
-            for (auto& remain_rpc : _num_in_flight_rpcs) {
-                std::lock_guard<Mutex> l(*_mutexes[remain_rpc.first]);
-                if (remain_rpc.second > 0) {
-                    ss << (remain_rpc_num > 0 ? ", " : "") << print_id(_instance_id2finst_id[remain_rpc.first]);
-                    remain_rpc_num++;
+        if (state != nullptr && state->query_ctx()->is_query_expired()) {
```
Comment (liuyehcf) [inline be/src/exec/pipeline/exchange/sink_buffer.cpp:197]: what if the query is not expired? we don't need to log?

File: be/src/runtime/sender_queue.cpp (hunk 5 above)
Comment (fzhedu) [reply be/src/runtime/sender_queue.cpp:540]: if the if is reached, there are bugs, should print a log to show this problem.

File: be/src/runtime/data_stream_mgr.cpp (hunk 3 above)
Comment (fzhedu) [reply be/src/runtime/data_stream_mgr.cpp:154]: help us to know the exchange receiver is alive longer than sender.

File: be/src/runtime/sender_queue.cpp (hunk 4 above)
Comment (fzhedu) [reply be/src/runtime/sender_queue.cpp:670]: ok, I will change it to VLOG, as `_num_remaining_senders <= 0` is checked above.

File: be/src/exec/pipeline/exchange/sink_buffer.cpp (hunk 6 above)
Comment (fzhedu) [reply be/src/exec/pipeline/exchange/sink_buffer.cpp:197]: If the query is cancelled by users or other errors, the root causes may not from exchange. 
just if the query is blocked by exchange, we should know these msg.
if other cases need this logs, we can open it later.

**Review by liuyehcf — APPROVED** [review]

File: be/src/exec/pipeline/scan/scan_operator.cpp (hunk 2 above)
Comment (satanson) [reply be/src/exec/pipeline/scan/scan_operator.cpp:222]: can we turn on verbose flag in runtime when we need it ?

File: be/src/runtime/sender_queue.cpp (hunk 5 above)
Comment (stdpain) [reply be/src/runtime/sender_queue.cpp:540]: It's not a bug.

**Comment by fzhedu** [comment]
https://github.com/StarRocks/starrocks/pull/36836

**Pull request merged**